formidling/src/main/java/no/nav/ung/sak/formidling/innhold/InnvilgelseInnholdBygger.java

```diff
@@ -0,0 +1,195 @@
+            .getFødselsdato();
+        boolean oppnårMaksAlder = vurderAntallDagerResultat.tidslinjeNokDager()
+            .getMaxLocalDate()
+            .isAfter(fødselsdato.plusYears(Sats.HØY.getTomAlder()));
+
+        return new ResultatFlaggDto(
+            enDagsats,
+            ettGbeløp,
+            lavSats,
+            høySats,
+            varierendeSats,
+            oppnårMaksAlder);
+    }
+
+    private static Set<GbeløpPeriodeDto> lagGbeløpPerioderDto(LocalDateTimeline<GrunnlagOgTilkjentYtelse> grunnlagOgTilkjentYtelseTimeline) {
+        return grunnlagOgTilkjentYtelseTimeline
+            .mapSegment(GrunnlagOgTilkjentYtelse::grunnbeløp)
+            .compress().stream()
+            .map(it -> new GbeløpPeriodeDto(
+                new PeriodeDto(it.getFom(), it.getTom()), it.getValue()))
+            .collect(Collectors.toSet());
+    }
+
+    private static SatserDto lagSatsDto(LocalDateTimeline<GrunnlagOgTilkjentYtelse> grunnlagOgTilkjentYtelseTimeline) {
+        List<GrunnlagOgTilkjentYtelse> sortertGrunnlagOgTilkjentYtelseVerdier = grunnlagOgTilkjentYtelseTimeline.stream()
+            .sorted(Comparator.comparing(LocalDateSegment::getLocalDateInterval, Comparator.reverseOrder())) // nyeste først
+            .map(LocalDateSegment::getValue)
+            .distinct()
+            .toList();
+
+        var nyesteLavSats = sortertGrunnlagOgTilkjentYtelseVerdier.stream()
+            .filter(it -> it.satsType() == UngdomsytelseSatsType.LAV)
+            .findFirst()
+            .map(GrunnlagOgTilkjentYtelse::grunnbeløpFaktor);
+
+        var nyesteHøySats = sortertGrunnlagOgTilkjentYtelseVerdier.stream()
+            .filter(it -> it.satsType() == UngdomsytelseSatsType.HØY)
+            .findFirst()
+            .map(GrunnlagOgTilkjentYtelse::grunnbeløpFaktor);
+
+        return new SatserDto(nyesteHøySats.orElse(null), nyesteLavSats.orElse(null), Sats.LAV.getTomAlder(), Sats.HØY.getTomAlder());
+    }
+
+    private static List<TilkjentPeriodeDto> lagTilkjentePerioderDto(LocalDateTimeline<GrunnlagOgTilkjentYtelse> grunnlagOgTilkjentYtelseTimeline) {
+        return grunnlagOgTilkjentYtelseTimeline
+            .mapSegment(it ->
+                new TilkjentYtelseDto(it.dagsats(), it.grunnbeløpFaktor(), it.grunnbeløp(), it.årsbeløp()))
+            .compress().stream()
+            .sorted(Comparator.comparing(LocalDateSegment::getLocalDateInterval)) //eldste først for tilkjent perioder
+            .map(it ->
+                new TilkjentPeriodeDto(new PeriodeDto(it.getFom(), it.getTom()), it.getValue()))
+            .toList();
+    }
+
+    private static LocalDateSegmentCombinator<DagsatsOgUtbetalingsgrad, UngdomsytelseSatser, GrunnlagOgTilkjentYtelse> sammenstillGrunnlagOgTilkjentYtelse() {
+        return (di, lhs, rhs) -> {
+            var dg = lhs.getValue();
+            var sp = rhs.getValue();
+            return new LocalDateSegment<>(di,
+                new GrunnlagOgTilkjentYtelse(
```

Comment:
Er det relevant å mappe avslagsårsak fra uttak også? Per no brukes det dersom man ikkje har fleire dager igjen, men det er også tenkt brukt dersom man har for høg inntekt

Reply:
godt innspill. Er det bare da den blir brukt og ikke ved programslutt ? 
Kan ta det når det kommer scenario på det